worker/dependency/util.go

```diff
@@ -100,3 +101,20 @@ func SelfManifold(engine *Engine) Manifold {
 		},
 	}
 }
+
+// ManifoldDependencies returns all manifold dependencies.
+func (all Manifolds) ManifoldDependencies(name string, manifold Manifold) set.Strings {
+	if len(manifold.Inputs) == 0 {
+		return set.NewStrings(name)
```

Comment:
Is this right? It seems odd - if a manifold has no dependencies then it depends on itself? And it seems like if you want to do that then you'd also do it in the case where there are dependencies.

Reply:
Well, this is a recursive function. So this is actually, the end of the recursion, the deepest, darkest hours (or lightest, depending on whether you half a half-full- or half-empty-glass day...)